torch/nn/attention/_flex_attention.py

```diff
@@ -289,9 +299,13 @@ def _create_block_mask(
         block_mask (tuple): A tuple of (kv_num_blocks, kv_indices, q_num_blocks, q_indices,
                             KV_BLOCK_SIZE, Q_BLOCK_SIZE) which represents the block mask.
     """
-    mask = _create_mask(score_mod, B, H, M, N, device)
-    block_mask = _create_block_mask_from_mask(mask, KV_BLOCK_SIZE, Q_BLOCK_SIZE)
-    return block_mask
+    inner_func = _create_block_mask_inner
+    # This is kind of a temporary hack to workaround some issues
+    if _compiled:
+        inner_func = torch.compile(inner_func)
+
+    with TransformGetItemToIndex():
```

Comment:
I need to move this `TransformGetItemToIndex` (a TorchFunctionMode) out of the inner function and then create another inner function that's wrapped in order to compile this.

cc: @zou3519 @yanboliang